modules/dag-shared/src/main/scala/org/tessellation/dag/snapshot/SnapshotOrdinal.scala

```diff
@@ -4,18 +4,17 @@ import cats.Order
 import cats.kernel._
 import cats.syntax.semigroup._
 
-import org.tessellation.schema.{nonNegLongDecoder, nonNegLongEncoder}
-
 import derevo.cats.{order, show}
-import derevo.circe.magnolia.{decoder, encoder}
 import derevo.derive
 import eu.timepit.refined.auto._
 import eu.timepit.refined.cats._
 import eu.timepit.refined.numeric.NonNegative
 import eu.timepit.refined.refineV
 import eu.timepit.refined.types.numeric.NonNegLong
+import io.circe.refined._
+import io.circe.{Decoder, Encoder}
 
-@derive(encoder, decoder, order, show)
+@derive(order, show)
 case class SnapshotOrdinal(value: NonNegLong)
```

Comment:
btw, why we don't have `newtype` here? 🤔 
@kpudlik @TheMMaciek @tbekas maybe you remember the reason?

Reply:
tl;dr; - it was required for Consensus (ClassTag / AnyRef). Probably it can be a newtype now, but it requires a schema migration